packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -109,6 +109,14 @@ export class UPlotChart extends React.Component<PlotProps, UPlotChartState> {
       this.reinitPlot();
     } else if (!sameData(prevProps, this.props)) {
       ctx.plot?.setData(this.props.data);
+
+      // this is a uPlot cache-busting hack for bar charts in case x axis labels changed
+      // since the x scale's "range" doesnt change, the axis size doesnt get recomputed, which is where the tick labels are regenerated & cached
+      // the more expensive, more proper/thorough way to do this is to force all axes to recalc: plot?.redraw(false, true);
+      if (ctx.plot && typeof this.props.data[0][0] === 'string') {
+        //@ts-ignore
+        plot.axes[0]._values = this.props.data[0];
```

Comment:
plot isn't defined in 8.1.x, so this causes an undefined error

Reply:
ah shit, it should have been `ctx.plot`

Reply:
yeah, and it wasn't picked up due to the ts-ignore comment :(

Reply:
that has to be there unfotunately because `_values` is not a public API and is not in the TS interface.

i also couldn't get the 8.1.x branch building due to cached `wire` gen failing on `make run`. apparently i needed to delete it?